pre_commit/languages/python.py

```diff
@@ -196,7 +196,14 @@ def install_environment(
     python = norm_version(version)
     if python is not None:
         venv_cmd.extend(('-p', python))
-    install_cmd = ('python', '-mpip', 'install', '.', *additional_dependencies)
+    install_cmd = (
+        'python',
+        '-mpip',
+        'install',
+        '--isolated',
```

Comment:
this will break people who depend on pip configuration variables, especially `PIP_INDEX_URL` (a common enterprise setup)

Reply:
if not the change in this PR, what is recommended for enterprise setups that use pip.ini file to specify extra index url?

another alternative - perhaps pre-commit's config (don't know where) can allow a user to specify extra args for the pip call?

Reply:
are you typing passwords every time you install? should you put an api key in your pip configuration?

Reply:
> are you typing passwords every time you install

no, I use `keyring` package to manage the passwords/tokens/etc.

> should you put an api key in your pip configuration?

i'm not a fan of having username/passwords in pip.ini file in plain text format. another solutipon would be to store them in ~/.netrc file but it's still plain-text-storage-on-disk. while keyring at least tries to hide things in some decent way.

Reply:
@asottile, it doesn't seem like there is a satisfying solution to this problem. Therefore, I do not understand your premature closing of this PR.

Reply:
`--isolated` is not a solution, therefore I closed the PR -- I'm not sure why you think that's premature?

Reply:
> I'm not sure why you think that's premature?

because it's a real problem :)

however, i think the salomon solution would be to let pre-commit pass some extra args to how it calls pip internally. that way folks like me will be able to pass --isolated, and some other folks will be able to NOT pass it in order to use `PIP_INDEX_URL` or whatever else there is. is this a decent approach?

Reply:
> > I'm not sure why you think that's premature?
>
> because it's a real problem :)

well this PR is not a solution => therefore closed.

> that way folks like me will be able to pass --isolated

can't you already do that by setting `PIP_ISOLATED=1`? or `additional_dependencies: [--isolated]`?

Reply:
> well this PR is not a solution => therefore closed

fair.

> can't you already do that by setting PIP_ISOLATED=1? or additional_dependencies: [--isolated]?

`additional_dependencies: [--isolated]` would be great, but it didn't work (i'll try again to confirm that i did it right). and setting an environment variable only to run a single command does not seem sustainable for me.

Reply:
if it didn't work in `additional_dependencies` then this patch won't work, they're exactly the same thing